File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.java
class Solution {
    public int bestClosingTime(String customers) {
        int cur_penalty = 0;
        for(int i=0; i<customers.length(); i++){
            if(customers.charAt(i) == 'Y'){
                cur_penalty++;
            }
        }
        int min_penalty = cur_penalty;
        int early = 0;
        for(int i=0; i<customers.length(); i++){
            char ch = customers.charAt(i);
            if(ch == 'Y'){
                cur_penalty--;
            }
            else{
                cur_penalty++;
            }

            if(cur_penalty < min_penalty){
                early = i+1;
                min_penalty = cur_penalty;
            }
        }
        return early;
    }
}

File: 2483-minimum-penalty-for-a-shop/2483-minimum-penalty-for-a-shop.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        max_score = 0
        ans = -1
        curr = 0
        for i, ch in enumerate(customers):
            if ch == 'Y':
                curr += 1
            else:
                curr -= 1
            
            if curr > max_score:
                ans = i
                max_score = curr
        
        return ans + 1